File: OfficeCelebr8/OfficeCelebr8.Domain/Models/Room.cs
using System.ComponentModel.DataAnnotations;

namespace OfficeCelebr8.Domain.Models
{
    public class Room
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }
        public string Description { get; set; }
        public int Capacity { get; set; } = 2;

        [Required]
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.Now;

        [Required]
        public DateTime EventOn { get; set; }

        [Required]
        public int TotalCollection { get; set; }
        public int CollectedTillNow { get; set; } = 0;

        // Navigation property - One Room has many RoomMembers
        public ICollection<RoomMember> Members { get; set; } = new List<RoomMember>();
    }
}

File: OfficeCelebr8/OfficeCelebr8.Frontend/src/app/models/room/room.ts
export class Room {
    id : number;
    name : string;
    description : string;
    capacity : number;
    createdBy : number;
    createdOn : Date;
    eventOn : Date;
    totalCollection : number;
    collectedTillNow : number;
    members : string;
    constructor(id : number, name : string, description : string, capacity : number, createdBy : number,
                createdOn : Date, eventOn : Date, totalCollection : number, collectedTillNow : number,
                members : string)
    {
        this.id = id;
        this.name = name;
        this.description = description;
        this.capacity = capacity;
        this.createdBy = createdBy;
        this.createdOn = createdOn;
        this.eventOn = eventOn;
        this.totalCollection = totalCollection;
        this.collectedTillNow = collectedTillNow;
        this.members = members;
    }
}
